src/parser/jobs/sch/modules/DoTs.js

```diff
@@ -28,12 +29,146 @@ export default class DoTs extends CoreDoTs {
 		...DoTs.dependencies,
 		'checklist',
 		'suggestions',
+		'downtime',
 	]
 
 	static statusesToTrack = [
 		STATUSES.BIOLYSIS.id,
 	]
 
+	_lastBioCast = undefined
+	_application = {}
+
+	constructor(...args) {
+		super(...args)
+
+		const castFilter = {
+			by: 'player',
+			abilityId: ACTIONS.BIOLYSIS.id,
+		}
+		this.addEventHook('cast', castFilter, this._onTrackedDotCast)
+		const statusFilter = {
+			by: 'player',
+			abilityId: [STATUSES.BIOLYSIS.id],
+		}
+		this.addEventHook(['applydebuff', 'refreshdebuff'], statusFilter, this._onTrackedDotApply)
+	}
+
+	_createTargetApplicationList() {
+		return {
+			[STATUSES.BIOLYSIS.id]: [],
+		}
+	}
+
+	_pushApplication(targetKey, statusId, event) {
+		const target = this._application[targetKey] = this._application[targetKey] || this._createTargetApplicationList()
+		const source = this._lastBioCast
+		target[statusId].push({event, source})
+	}
+
+	_onTrackedDotCast(event) {
+		this._lastBioCast = event.ability.guid
+	}
+
+	_onTrackedDotApply(event) {
+		const statusId = event.ability.guid
+
+		// Make sure we're tracking for this target
+		const applicationKey = `${event.targetID}|${event.targetInstance}`
+		//save the application for later use in the output
+		this._pushApplication(applicationKey, statusId, event)
+	}
+
+	_formatDriftText(timestamp, previous=0) {
+		let drift = 0
+		const lateAllow = 1000
+		const earlyAllow = -2500
+		if (previous !== 0) {
+			const delta = timestamp - previous
+			drift = delta - (STATUSES.BIOLYSIS.duration * 1000)
+			if (drift > (STATUSES.BIOLYSIS.duration * 1000)) {
+				const windows = this.downtime.getDowntimeWindows(timestamp - (STATUSES.BIOLYSIS.duration * 1000))
+				if (windows.length > 0) {
+					drift = 0
+				}
+			}
+			if (drift === 0) {
+				return <Trans id="sch.dots.drift.application">reapplication</Trans>
+			}
+			if (drift > lateAllow) {
+				return <Trans id="sch.dots.drift.late">{this.parser.formatDuration(Math.abs(drift))} late</Trans>
+			}
+			if (drift < earlyAllow) {
+				return <Trans id="sch.dots.drift.early">{this.parser.formatDuration(Math.abs(drift))} early</Trans>
+			}
+		}
+		return <Trans id="sch.dots.drift.application">normal application</Trans>
```

Comment:
Since you're adding new translations, you need to run `yarn extract` so that other languages are aware of the new strings that need translating.